### ISSUE TITLE
kWh sensor out by factor of 10

### ISSUE THREAD

**Issue by craigrallen** [StyraHem/ShellyForHASS#168]
Hey, the new 1.6.0 beta 9 version is excellent. Keep up the great work.

Just you let you know the sensor is saying 9.78kwh used for a unit in HA, but in the shelly app the same unit (shelly 2) is reporting 978wh which is .978kWh 

Also noticed the instant power consumption sensor no longer available. Only the total consumed power sensor showing. I guess there's meant to be two sensors? One that is instant power and one total consumed energy?

**Comment by hakana** [comment]
Ok, thanks letting me know!

**Comment by tomekswx** [comment]
In my case 40161.65 kWh reported by HA, 40.04 kWh reported by shellly app (correct value)

**Comment by chemelli74** [comment]
Hi, 

can you please check http://x.x.x.x/status ?
You'll find 2 "total" values, one per channel for a Shelly 2.5.

It doesn't seems to be a factor of 10 at all. 
I bet it depends on the uptime of the single unit or similar as the factor change a lot from one to another.

Simone

**Comment by morph027** [comment]
```
$ curl http://shelly4pro-cc958c/status | jq '.meters[] | .total'
3053
2011
4351
4621
```

![Screenshot from 2020-01-26 20-14-07](https://user-images.githubusercontent.com/600106/73140396-a8c5a600-4078-11ea-94f6-6ae17ff31b12.png)

**Comment by hakana** [comment]
FYI: The unit in HA is wrong, the value is Wh not kWh, but the calculation is correct

**Comment by hakana** [comment]
When you look in the app it shows the consumption for the period of the graph, 24h as default.

**Comment by craigrallen** [comment]
This is the status output from the shelly 

`{"wifi_sta":{"connected":true,"ssid":"ASUS_90_2G","ip":"192.168.255.196","rssi":-43},"cloud":{"enabled":true,"connected":true},"mqtt":{"connected":false},"time":"13:37","serial":22,"has_update":false,"mac":"CC50E3134F0D","relays":[{"ison":false,"has_timer":false,"overpower":false,"is_valid":true},{"ison":false,"has_timer":false,"overpower":false,"is_valid":true}],"rollers":[{"state":"stop","power":0.00,"is_valid":true,"safety_switch":false,"stop_reason":"normal","last_direction":"stop","current_pos":101,"calibrating":false,"positioning":true}],"meters":[{"power":0.00,"current":1.00,"voltage":14.00,"is_valid":true,"timestamp":1580218633,"counters":[0.000, 0.000, 0.000],"total":889}],"inputs":[{"input":0},{"input":0}],"update":{"status":"idle","has_update":false,"new_version":"20191216-090122/v1.5.7@c30657ba","old_version":"20191216-090122/v1.5.7@c30657ba"},"ram_total":50216,"ram_free":35396,"fs_size":233681,"fs_free":160138,"voltage":14.62,"uptime":265407}`

This is the current output in HA

![image](https://user-images.githubusercontent.com/56693437/73264663-9318c380-41d3-11ea-9530-732ea1c18622.png)

**Comment by tomekswx** [comment]
I have shelly plug, in my case it is factor 1000, but i have quick fix for this:

- platform: template
    sensors:
rekuperator_total_kwh:
        unit_of_measurement: 'kWh'
        entity_id:
          - sensor.shelly_shplg2_1_xxxxxx_total_consumption_attr
        value_template: >-
          {% set vall2 = states.sensor.shelly_shplg2_1_xxxxxx_total_consumption_attr.state|float %}
          {{ (vall2/1000)|round(1) }}

![obraz](https://user-images.githubusercontent.com/59449391/73266699-b6457200-41d7-11ea-82aa-473ed9009799.png)

**Comment by chemelli74** [comment]
@craigrallen, @morph027, @tomekswx please recheck with final 0.1.6 and report back.

Simone

**Comment by morph027** [comment]
Looked good in latest beta, installed final 0.1.6 today and it still looks good :+1:

**Comment by Xylomid** [comment]
Hello, my consumption sensors are still off by about factor 20 compared to shelly.cloud data. I'm using shellys EM. Power sensors are working great and i'm getting the right consumption by using the integration platform on power sensor.  Thanks

**Comment by hakana** [comment]
@Xylomid Can you send a screen capture of the cloud data and in HA. Is HA showing a bigger value?

**Comment by Xylomid** [comment]
![image](https://user-images.githubusercontent.com/53476963/73922324-4155f480-48c9-11ea-8bee-c60f883c3422.png)
![image](https://user-images.githubusercontent.com/53476963/73922332-45821200-48c9-11ea-8a70-b7116090047f.png)

Shelly cloud gives me 7147 Wh from 00am to 10am  
HA sensors gives me 121 Wh from 00am to 10am

**Comment by hakana** [comment]
Ok, look like Shelly EM report Wh, not Wmin as other devices. It is a factor of 60 and that gives you 7260 instead of 121.

**Comment by Xylomid** [comment]
That's it ! didn't think about that, thank you

**Comment by chemelli74** [comment]
Sorry, but I don't get it :-(

Looking at Shelly's app, at 8:00am of Feb 6 I see: 65Wh.
Looking at HA at 8:00am of Feb 6 I see: current total 891 (upper right corner) - at time total 747 (red line graph) = 144.

Where am I wrong ?

Simone

![2020-02-06 23_30_49-Window](https://user-images.githubusercontent.com/57354320/73984469-fd9ad380-4938-11ea-92e2-29fcff51e9d9.png)



![IMG_2440](https://user-images.githubusercontent.com/57354320/73984236-88c79980-4938-11ea-8eec-3ceca71ad178.PNG)

**Comment by Xylomid** [comment]
Looking at Shelly's app, at 8:00am of Feb 6 I see: 65Wh.
==> this is your energy consumption from 8am to 9am
Looking at HA at 8:00am of Feb 6 I see: current total 891 (upper right corner) - at time total 747 (red line graph) = 144.
==> 144wh is your energy consumption from 8am to now (end of graph)

**Comment by hakana** [comment]
I think everything is fine in the 0.1.6 except for the Shelly EM that showing wrong by a factor of 60. That will be fixed in the next release. Let me know if some other devices have a problem.

**Comment by chemelli74** [comment]
> I think everything is fine in the 0.1.6 except for the Shelly EM that showing wrong by a factor of 60. That will be fixed in the next release. Let me know if some other devices have a problem.

Dimitar is looking at it from the firmware point of view no later than tomorrow morning.

Simone

**Comment by hakana** [comment]
@chemelli74 I have fixed this in 0.1.7, let me know if any news from Dimitar.

**Comment by catalinbordan** [comment]
> @chemelli74 I have fixed this in 0.1.7, let me know if any news from Dimitar.

looking forward to see it 0.1.7, because it is not public to test it.
anyhow with the values in wh in HA I can not find a way to check it with shelly cloud.
can we have a posibility/option to convert it to kWh? plus we can use it as a source for utility meter.

**Comment by chemelli74** [comment]
> @chemelli74 I have fixed this in 0.1.7, let me know if any news from Dimitar.

Dimitar reported that both ShellyEM and Shelly3EM will report Wh; it's not a bug, it's a by-design behaviour.

As you have changed code already, will close this issue.

Simone
